examples/gen_nifti/main.rs

```diff
@@ -0,0 +1,29 @@
+//! An application for writing a NIFTI file from scratch
+
+extern crate nifti;
+
+use std::env;
+
+use crate::nifti::{writer::WriterOptions, Extender, Extension, ExtensionSequence};
+
+fn main() {
+    let mut args = env::args().skip(1);
+    let filename = args.next().expect("Path to NIFTI file is required");
+
+    // generate some test data 256x256 float32
+    let data = ndarray::Array3::<f32>::zeros((256, 256, 1));
+
+    let extension1 = Extension::new(8 + 4, 3, vec![0, 0, 0, 0]);
```

Comment:
@Enet4 I think our API could be greatly improved by removing the `esize: i32` argument.

    pub fn new(esize: i32, ecode: i32, edata: Vec<u8>) -> Self {
        if esize as usize != 8 + edata.len() panic!

This clearly indicates that we already know the answer and won't accept any other, so why ask it?

Reply:
Hmm true, the constructor is practically already encoding the extension size through `edata`, and we can reconstruct the `esize` field from that. A `new` function that does not ask for an `esize` would be easier to use and less error prone.